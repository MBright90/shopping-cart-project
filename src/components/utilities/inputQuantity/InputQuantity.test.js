import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'

import InputQuantity from './index'

describe('InputQuantity', () => {
  let props

  beforeEach(() => {
    props = {
      quantity: 3,
      setQuantity: jest.fn()
    }
  })

  it('renders to match snapshot', () => {
    const { container } = render(<InputQuantity {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('calls set quantity when plus button is clicked', () => {
    render(<InputQuantity {...props} />)

    const increaseQuantityButton = screen.getByText('+')
    userEvent.click(increaseQuantityButton)
    expect(props.setQuantity).toHaveBeenCalled()
  })

  it('calls set quantity when minus button is clicked', () => {
    render(<InputQuantity {...props} />)

    const decreaseQuantityButton = screen.getByText('-')
    userEvent.click(decreaseQuantityButton)
    expect(props.setQuantity).toHaveBeenCalled()
  })
})
