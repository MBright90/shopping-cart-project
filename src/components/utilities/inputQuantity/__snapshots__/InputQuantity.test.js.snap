// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputQuantity renders to match snapshot 1`] = `
<div>
  <div
    class="inputQuantity"
  >
    <button
      class="quantityButton"
      type="button"
    >
      -
    </button>
    <input
      class="quantityInput"
      data-testid="quantity-input"
      id="quantity-input"
      max="9"
      min="0"
      readonly=""
      type="number"
      value="3"
    />
    <button
      class="quantityButton"
      type="button"
    >
      +
    </button>
  </div>
</div>
`;
